src/flb_output.c

```diff
@@ -204,7 +204,12 @@ int flb_output_task_flush(struct flb_task *task,
         }
 
         flb_task_users_inc(task);
-        flb_coro_resume(out_flush->coro);
+        ret = flb_pipe_w(config->ch_self_events[1], &out_flush,
+                        sizeof(struct flb_output_flush*));
+        if (ret == -1) {
+            flb_errno();
+            return -1;
+        }
```

Comment:
Not sure I like calling flb_output_flush_create during flush, since this runs the initialization code which may be time and resource consuming. Might be better to create an engine task struct with the arguments and differ everything to low priority event. More complex though.